cpp/src/io/parquet/writer_impl.cu

```diff
@@ -981,7 +1031,9 @@ auto build_chunk_dictionaries(hostdevice_2dvector<gpu::EncColumnChunk>& chunks,
   std::vector<rmm::device_uvector<gpu::slot_type>> hash_maps_storage;
   hash_maps_storage.reserve(h_chunks.size());
   for (auto& chunk : h_chunks) {
-    if (col_desc[chunk.col_desc_id].physical_type == Type::BOOLEAN) {
+    if (col_desc[chunk.col_desc_id].physical_type == Type::BOOLEAN ||
+        (col_desc[chunk.col_desc_id].output_as_byte_array &&
+         col_desc[chunk.col_desc_id].physical_type == Type::BYTE_ARRAY)) {
```

Comment:
Are these two conditions maybe redundant?

Reply:
I went down this rabbit hole on the last one and list tests started failing, making me think that lists were getting into here in some way. I'm happy to try and resolve that, but it will take some time.

Reply:
Let's leave a TODO somewhere to clean up these checks (there's more seemingly redundant checks above).

Reply:
https://github.com/rapidsai/cudf/issues/11394